internal/controller/nopresource/nopresource.go

```diff
@@ -217,7 +189,29 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
 
 	fmt.Printf("Deleting: %+v", cr)
 
-	cr.Status.SetConditions(runtimev1alpha1.Deleting())
+	cr.Status.SetConditions(xpv1.Deleting())
 
 	return nil
 }
+
+func ReconcileLogic(conditionAfter []v1alpha1.ResourceConditionAfter, t time.Duration) []int {
+	latestTime := make(map[string]time.Duration)
+	latestIdx := make(map[string]int)
+	for i := 0; i < len(conditionAfter); i++ {
```

Comment:
this function looks like it's logically correct, but it could use a bit of clean up in the form of comments and variable names.  e.g., the function header should be commented to be clear about what the arguments mean, how they should be used, and what the return value means (a slice of ints that represents what exactly).  Some of the checks within the body of the function could use a comment saying in plain english what it is checking for.

The variable names could be cleaned up (more descriptive names) as i was getting a bit confused between `t`, `mx` and `elapsed`.